### PR TITLE
Fix gen_python_files usage for black>=21.7b1.dev9

### PR DESCRIPTION
**Pull request by flying-sheep**
This should fix the tests and be backwards compatible

### CODE REVIEW

**Comment by akaihola** [comment]
@flying-sheep oops, I already had a fix for this in #183. The difference is that my solution looks at `black.__version__` and you're inspecting the actual signature of the function. Is this just a matter of taste or do you see pros and cons in each approach?

File: src/darker/black_diff.py [hunk 1]
```diff
@@ -131,6 +131,9 @@ def apply_black_excludes(
     :return: Absolute paths of files which should be reformatted using Black
 
     """
+    sig = inspect.signature(gen_python_files)
+    # those two exist and are required in black>=21.8b0
```
Comment (akaihola) [inline src/darker/black_diff.py:135]: They were actually introduced in `21.7b1.dev9`.

File: src/darker/black_diff.py [hunk 2]
```diff
@@ -131,6 +131,9 @@ def apply_black_excludes(
...
+    kwa = dict(verbose=False, quiet=False) if "verbose" in sig.parameters else {}
```
Comment (akaihola) [inline src/darker/black_diff.py:136]: Just a stylistic thing, I'd use the more commonly seen name `kwargs` for the extra keyword arguments.
```suggestion
    kwargs = dict(verbose=False, quiet=False) if "verbose" in sig.parameters else {}
```

File: src/darker/black_diff.py [hunk 3]
```diff
@@ -141,6 +144,7 @@ def apply_black_excludes(
             force_exclude=black_config.get("force_exclude"),
             report=Report(),
             gitignore=None,
+            **kwa,
```
Comment (akaihola) [inline src/darker/black_diff.py:147]: ...here, too
```suggestion
            **kwargs,
```

**Comment by akaihola** [comment]
Could you add a change log entry similar to [the one in the alternative pull request](https://github.com/akaihola/darker/pull/183/files#diff-ff3c479edefad986d2fe6fe7ead575a46b086e3bbcf0ccc86d85efc4a4c63c79R15)?

**Review by akaihola — CHANGES_REQUESTED** [review]
You can see a couple of comments in the review. I'm starting to think I like the `inspect` approach better than looking at the Black version.

**Comment by flying-sheep** [comment]
> Is this just a matter of taste or do you see pros and cons in each approach?

I think the approach here is simpler in two ways:

1. it uses less code
2. it checks directly for the difference that has to be handled as opposed to taking an indirect route over the version

but ultimately it’s a matter of taste as both solutions are likely to be robust.

**Review by akaihola — APPROVED** [review]
Thanks @flying-sheep for the improved solution!

**Pull request merged**